test/StateDiffChecker.t.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {Test} from "forge-std/Test.sol";
+import {VmSafe} from "forge-std/Vm.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {LibStateDiffChecker as Checker} from "script/LibStateDiffChecker.sol";
+
+/// @dev A simple contract to which makes some state modifications for the purpose of testing state diff checking.
+contract Target {
+    uint256 public x = 1;
+    address public y;
+    mapping(uint256 => bytes32) public z;
+
+    Callee public callee;
+    DelegateCallee public delegateCallee;
+
+    constructor() {
+        callee = new Callee();
+        delegateCallee = new DelegateCallee();
+    }
+
+    function set() external {
+        x = 0;
+        y = address(0xabba);
+        z[1] = hex"acdc";
+
+        callee.set1();
+        (bool success,) =
+            address(callee).delegatecall(abi.encodeWithSelector(callee.set2.selector, address(delegateCallee)));
+        require(success, "Target: Delegatecall failed");
+    }
+}
+
+contract Callee {
+    function set1() external {
+        assembly {
+            sstore(0x1111, 1)
+        }
+    }
+
+    function set2(DelegateCallee delegateCallee) external {
+        (bool success,) = address(delegateCallee).delegatecall("");
+        require(success, "Callee: Delegatecall failed");
+    }
+}
+
+contract DelegateCallee {
+    fallback() external {
+        assembly {
+            sstore(0x2222, 1)
+        }
+    }
+}
+
+/// @dev Tests for LibStateDiffChecker
+contract StateDiffChecker_Test is Test {
+    Target public target;
+
+    error StateDiffMismatch(string field, bytes32 expected, bytes32 actual);
+
+    function setUp() public {
+        target = new Target();
+    }
+
+    /// @dev Test that the sample testDiff.json file is read and parsed correctly
+    function test_parseDiffSpecs() public {
+        string memory _path = "test/testDiff.json";
+        string memory json = vm.readFile(_path);
+        Checker.StateDiffSpec memory diffSpec = Checker.parseDiffSpecs(json);
+
+        assertEq(diffSpec.chainId, 31337);
+        assertEq(diffSpec.storageSpecs.length, 3);
```

Comment:
This should be 5 now it seems—we should run these tests in CI

Reply:
Good catch. I added these to CI, and refactored the parsing test to check that the hash of the full diff matches.

Reply:
LGTM. Any thoughts on adding the `deny_warnings = true` to a CI profile also? We now have a solc warning in this method

Reply:
Fixed and added the ci profile: [`0d07195` (#66)](https://github.com/ethereum-optimism/superchain-ops/pull/66/commits/0d071951e0445a14083d5e48a222720ad32fc628)